tests/functional/test_ci.py

```diff
@@ -36,3 +41,24 @@ def test_version_option():
     assert not ret.stderr, "Nothing should be written to stderr"
     assert ret.returncode == 0, "A non-successful return code was provided"
     assert ret.stdout == expected_output, "Output did not match expected input"
+
+
+@pytest.mark.parametrize(
+    ("supplied_args", "expected_level"),
+    [
+        ([], logging.WARNING),
+        (["-v"], logging.INFO),
+        (["-vv"], logging.DEBUG),
+        (["-vvv"], LOGGING_TRACE_LEVEL),
+        (["-vvvv"], LOGGING_TRACE_LEVEL),
+        (["-q"], logging.ERROR),
+        (["-qq"], logging.CRITICAL),
+        (["-qqq"], logging.CRITICAL),
```

Comment:
Why are `-vvvv` and `-qqq` provided here when they're the same as `-vvv` and `-qq`?

Reply:
If a user puts in more than the maximum support verbosity/quiet options, it should not error but instead simply use that maximum level. These are test cases to ensure that happens.